core/src/main/java/io/ultrabrew/metrics/util/DistributionBucket.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics.util;
+
+import static io.ultrabrew.metrics.util.Commons.checkArgument;
+
+import java.util.function.Predicate;
+
+public class DistributionBucket {
+
+  private static final String UNDERFLOW = "underflow";
+  private static final String OVERFLOW = "overflow";
+
+  private final long[] buckets;
+
+  public DistributionBucket(final long[] buckets) {
+
+    checkArgument(buckets.length > 1, "Minimum bucket length is 2");
+    checkArgument(isSorted(buckets), "Bucket should be sorted in ascending order");
+    checkArgument(!hasDuplicate(buckets), "Bucket should not have duplicate entries");
+
+    this.buckets = buckets.clone();
+  }
+
+  public int getCount() {
+    return buckets.length + 1; // includes the underflow and overflow buckets
+  }
+
+  public int getBucketIndex(long value) {
+    return binarySearch(value);
```

Comment:
This is a custom implementation where the entries in the array represents a range instead of an absolute value. Can't use `Arrays.binarySearch()`. It will be clear when I add the java doc.